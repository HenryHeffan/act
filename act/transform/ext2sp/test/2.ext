timestamp 1536774731
version 8.1
tech scmos
style HP0.5um(hpcmos14tb)from:T24L
scale 1000 1 30
resistclasses 2600 2300 721000 721000 1 2300 2300 70 70 50
use nor2 nor2_0 1 0 8 0 1 24
use inv inv_0 1 0 34 0 1 10
use nor2 nor2_1 1 0 68 0 1 24
use inv inv_1 1 0 94 0 1 10
use inv inv_2 -1 0 149 0 1 10
use nor2 nor2_2 -1 0 175 0 1 24
node "m2_151_n6#" 0 47.748 151 -6 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "m2_116_n6#" 0 82.32 116 -6 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 33 28 0 0
node "m2_85_n6#" 0 47.748 85 -6 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "m2_25_n6#" 0 47.748 25 -6 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "m2_151_38#" 0 28.548 151 38 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "m2_116_38#" 0 30.744 116 38 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 33 28 0 0
node "m2_85_38#" 0 28.548 85 38 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "m2_25_38#" 0 28.548 25 38 m2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0
node "in3.t" 1 895.088 186 -13 v1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 100 72 16 16 0 0
node "in2" 1 694.664 154 -13 v1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 76 56 16 16 0 0
node "m1_151_14#" 0 210.741 151 14 m1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0 0 0
node "m1_85_7#" 2 2278.35 85 7 m1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 273 188 0 0 0 0
node "in1" 1 694.664 25 7 m1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 76 56 16 16 0 0
node "m1_85_14#" 0 210.741 85 14 m1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0 0 0
node "m1_25_14#" 0 210.741 25 14 m1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 21 20 0 0 0 0
node "in0" 1 895.088 -7 -13 v1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 100 72 16 16 0 0
node "out" 1 472.919 115 44 v1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 103 74 16 16 0 0
node "w_151_18#" 2369 1463.49 151 18 nw 0 0 0 0 161 60 0 0 0 0 0 0 0 0 0 0 0 0 0 0
node "w_116_18#" 1508 2299.77 116 18 nw 0 0 0 0 253 68 0 0 0 0 0 0 0 0 0 0 0 0 0 0
node "w_85_18#" 2369 1463.49 85 18 nw 0 0 0 0 161 60 0 0 0 0 0 0 0 0 0 0 0 0 0 0
node "w_25_18#" 2369 1463.49 25 18 nw 0 0 0 0 161 60 0 0 0 0 0 0 0 0 0 0 0 0 0 0
cap "m2_116_38#" "w_116_18#" 51.576
cap "m1_25_14#" "in1" 46.174
cap "m2_116_38#" "out" 152.425
cap "w_116_18#" "out" 440.634
cap "in2" "m1_151_14#" 46.174
cap "m1_85_7#" "out" 27.7044
cap "w_116_18#" "m1_85_7#" 36.843
cap "m2_151_38#" "w_151_18#" 45.129
cap "m2_85_n6#" "m1_85_7#" 120.609
cap "m2_116_n6#" "m1_85_7#" 101.475
cap "m1_85_7#" "m1_85_14#" 46.174
cap "m2_25_n6#" "in1" 188.542
cap "m2_25_38#" "w_25_18#" 45.129
cap "m2_85_38#" "w_85_18#" 45.129
cap "m2_151_n6#" "in2" 188.542
subcap "m2_85_38#" -62.84
subcap "m2_25_38#" -62.84
subcap "m1_85_7#" -434.733
cap "nor2_0/GND!" "in0" 139.904
cap "nor2_0/Vdd!" "nor2_0/w_n12_n6#" -3.12639e-13
cap "nor2_0/out" "in1" -2.13163e-14
cap "nor2_0/GND!" "nor2_1/b" 126.978
cap "nor2_1/out" "nor2_1/b" -2.13163e-14
cap "nor2_0/out" "nor2_0/w_n12_n6#" -2.27374e-13
cap "nor2_0/GND!" "in1" 126.979
subcap "m1_85_7#" -610.105
cap "inv_1/GND!" "in3.t" 139.904
cap "inv_2/out" "nor2_1/w_n12_n6#" -4.26326e-14
cap "out" "nor2_1/w_n12_n6#" 129.69
cap "inv_1/GND!" "in2" 126.979
cap "inv_1/Vdd!" "nor2_1/w_n12_n6#" -17.568
cap "in2" "inv_2/in" 3.55271e-14
cap "inv_2/in" "nor2_1/w_n12_n6#" -2.27374e-13
cap "inv_1/GND!" "inv_2/out" 80.6993
cap "out" "inv_1/Vdd!" 36.1173
merge "inv_2/out" "nor2_1/b" 35.364 0 0 0 0 0 0 0 0 0 0 0 0 0 0 24 -12 0 0 0 0
merge "nor2_1/b" "m1_85_7#"
merge "inv_0/in" "nor2_0/out" -70.74 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -12 0 0 0 0
merge "nor2_0/out" "m1_25_14#"
merge "nor2_2/GND!" "inv_2/GND!" -21.033 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 27 -24 0 0
merge "inv_2/GND!" "inv_1/GND!"
merge "inv_1/GND!" "nor2_1/GND!"
merge "nor2_1/GND!" "inv_0/GND!"
merge "inv_0/GND!" "nor2_0/GND!"
merge "nor2_0/GND!" "m2_25_n6#"
merge "m2_25_n6#" "m2_85_n6#"
merge "m2_85_n6#" "m2_116_n6#"
merge "m2_116_n6#" "m2_151_n6#"
merge "nor2_2/w_n12_n6#" "inv_2/w_n2_8#" 1881.63 0 0 0 0 207 -184 0 0 0 0 0 0 0 0 0 0 0 0 0 0
merge "inv_2/w_n2_8#" "inv_1/w_n2_8#"
merge "inv_1/w_n2_8#" "nor2_1/w_n12_n6#"
merge "nor2_1/w_n12_n6#" "inv_0/w_n2_8#"
merge "inv_0/w_n2_8#" "nor2_0/w_n12_n6#"
merge "nor2_0/w_n12_n6#" "w_25_18#"
merge "w_25_18#" "w_85_18#"
merge "w_85_18#" "w_116_18#"
merge "w_116_18#" "w_151_18#"
merge "inv_1/in" "nor2_1/out" 48.627 0 0 0 0 0 0 0 0 0 0 0 0 0 0 27 -12 0 0 0 0
merge "nor2_1/out" "m1_85_14#"
merge "nor2_2/Vdd!" "inv_2/Vdd!" 3.123 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 27 -24 0 0
merge "inv_2/Vdd!" "inv_1/Vdd!"
merge "inv_1/Vdd!" "nor2_1/Vdd!"
merge "nor2_1/Vdd!" "inv_0/Vdd!"
merge "inv_0/Vdd!" "nor2_0/Vdd!"
merge "nor2_0/Vdd!" "m2_25_38#"
merge "m2_25_38#" "m2_85_38#"
merge "m2_85_38#" "m2_116_38#"
merge "m2_116_38#" "m2_151_38#"
merge "nor2_2/out" "inv_2/in" -70.74 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -12 0 0 0 0
merge "inv_2/in" "m1_151_14#"
merge "nor2_0/a" "in0" -35.37 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
merge "nor2_0/b" "in1" -35.37 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
merge "nor2_2/b" "in2" -35.37 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
merge "nor2_2/a" "in3.t" -35.37 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
merge "nor2_1/a" "inv_0/out" -35.37 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
merge "inv_1/out" "out" -165.06 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -6 0 0 0 0
