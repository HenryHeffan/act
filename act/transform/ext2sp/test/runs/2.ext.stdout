*---------------------------------------------------
* Subcircuit from nor2.ext
*---------------------------------------------------
.subckt nor2 _
* -- connections ---
* -- fets ---
M1 a_1_0# a Vdd w_n12_n6# p W=3.3U L=0.6U
+ AS=4.95P PS=9.6U AD=2.97P PD=8.4U 
M2 out b a_1_0# w_n12_n6# p W=3.3U L=0.6U
+ AS=0P PS=0P AD=4.95P PD=9.6U 
M3 out a GND GND n W=1.2U L=0.6U
+ AS=3.6P PS=10.8U AD=2.16P PD=6U 
M4 GND b out GND n W=1.2U L=0.6U
+ AS=0P PS=0P AD=0P PD=0P 
* -- caps ---
C5 w_n12_n6# Vdd 0.427167F
C6 b a 0.131378F
C7 b w_n12_n6# 0.526042F
C8 b out 0.410568F
C9 a w_n12_n6# 0.627701F
C10 w_n12_n6# out 0.260849F
C11 a GND 1.00708F
C12 Vdd GND 0.134306F
C13 b GND 1.22139F
C14 out GND 0.698042F
C15 w_n12_n6# GND 6.06303F
.ends
*---------------------------------------------------
* Subcircuit from inv.ext
*---------------------------------------------------
.subckt inv _
* -- connections ---
* -- fets ---
M1 out in Vdd w_n2_8# p W=3.3U L=0.6U
+ AS=4.95P PS=9.6U AD=4.95P PD=9.6U 
M2 out in GND GND n W=1.2U L=0.6U
+ AS=1.8P PS=5.4U AD=1.8P PD=5.4U 
* -- caps ---
C3 out Vdd 0.115435F
C4 out w_n2_8# 0.300639F
C5 w_n2_8# Vdd 0.394932F
C6 w_n2_8# in 0.449068F
C7 w_n2_8# GND 5.01768F
C8 in GND 1.13587F
C9 Vdd GND 0.123326F
C10 out GND 0.544288F
.ends
*
*---------------------------------------------------
*  Main extract file 2.ext [scale=1]
*---------------------------------------------------
*
*--- subcircuits ---
xnor2_2 GND nor2
xinv_2 GND inv
xinv_1 GND inv
xnor2_1 GND nor2
xinv_0 GND inv
xnor2_0 GND nor2
* -- connections ---
V1 out xinv_1:out
V2 xinv_0:out xnor2_1:a
V3 in3.t xnor2_2:a
V4 in2 xnor2_2:b
V5 in1 xnor2_0:b
V6 in0 xnor2_0:a
V7 m1_151_14# xinv_2:in
V8 xinv_2:in xnor2_2:out
V9 m2_151_38# m2_116_38#
V10 m2_116_38# m2_85_38#
V11 m2_85_38# m2_25_38#
V12 m2_25_38# xnor2_0:Vdd!
V13 m1_85_14# xnor2_1:out
V14 xnor2_1:out xinv_1:in
V15 w_151_18# w_116_18#
V16 w_116_18# w_85_18#
V17 w_85_18# w_25_18#
V18 w_25_18# xnor2_0:w_n12_n6#
V19 xnor2_0:w_n12_n6# xinv_0:w_n2_8#
V20 xinv_0:w_n2_8# xnor2_1:w_n12_n6#
V21 xnor2_1:w_n12_n6# xinv_1:w_n2_8#
V22 xinv_1:w_n2_8# xinv_2:w_n2_8#
V23 xinv_2:w_n2_8# xnor2_2:w_n12_n6#
V24 m2_151_n6# m2_116_n6#
V25 m2_116_n6# m2_85_n6#
V26 m2_85_n6# m2_25_n6#
V27 m2_25_n6# xnor2_0:GND!
V28 m1_25_14# xnor2_0:out
V29 xnor2_0:out xinv_0:in
V30 m1_85_7# xnor2_1:b
V31 xnor2_1:b xinv_2:out
* -- caps ---
C32 w_25_18# w_25_18# 1.88163F
C33 out w_25_18# 0.12969F
C34 w_25_18# out 0.440634F
C35 Vdd out 0.152425F
C36 m1_85_14# GND 0.210741F
C37 m1_25_14# GND 0.210741F
C38 w_25_18# GND 6.69024F
C39 m1_151_14# GND 0.210741F
C40 Vdd GND 0.116388F
C41 in1 GND 1.01019F
C42 m1_85_7# GND 2.70811F
C43 in2 GND 1.01019F
C44 out GND 0.472919F
C45 in0 GND 1.03499F
C46 in3.t GND 1.03499F
*--- inferred globals
.global Vdd
.global GND
